sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfigurationprovider/_azureappconfigurationprovider.py

```diff
@@ -0,0 +1,183 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+
+from urllib.parse import urlparse
+import json
+import warnings
+from azure.appconfiguration import AzureAppConfigurationClient
+from azure.keyvault.secrets import SecretClient
+from ._settingselector import SettingSelector
+from ._user_agent import USER_AGENT
+
+
+class AzureAppConfigurationProvider:
+    """
+    Provides a dictionary-like interface to Azure App Configuration settings. Enables loading of sets of configuration
+    settings from Azure App Configuration into a Python application. Enables trimming of prefixes from configuration
+    keys. Enables resolution of Key Vault references in configuration settings.
+    """
+
+    def __init__(self):
+        # type: () -> None
+        self._dict = {}
+        self._trim_prefixes = []
+        self._client = None
+
+    @classmethod
+    def load(cls, connection_string=None, endpoint=None, credential=None, **kwargs):
+        """
+        Loads configuration settings from Azure App Configuration into a Python application.
+        :param connection_string: Connection string (one of connection_string or endpoint and credential must be set)
+        :type connection_string: str
+        :param endpoint: Endpoint (one of connection_string or endpoint and credential must be set)
+        :type endpoint: str
+        :param credential: Credential (one of connection_string or endpoint and credential must be set)
+        :type credential: Union[AppConfigConnectionStringCredential, TokenCredential]
+        :keyword selectors: List of setting selectors to filter configuration settings
+        :type selectors: list[~azure.appconfigurationprovider.SettingSelector]
+        :keyword trim_prefixes: List of prefixes to trim from configuration keys
+        :type trim_prefixes: list[str]
+        :keyword key_vault_options: Options for resolving Key Vault references
+        :type key_vault_options: ~azure.appconfigurationprovider.KeyVaultOptions
+        """
+        provider = AzureAppConfigurationProvider()
+
+        key_vault_options = kwargs.pop("key_vault_options", None)
+
+        provider.__buildprovider(connection_string, endpoint,
+                                 credential, key_vault_options)
+
+        selects = kwargs.pop("selects", {SettingSelector("*", "\0")})
+        provider._trim_prefixes = kwargs.pop("trimmed_key_prefixes", [])
+
+        provider._dict = {}
+        secret_clients = {}
+
+        for select in selects:
+            configurations = provider._client.list_configuration_settings(
```

Comment:
I am trying to think of the best way to do this, would we want to wrap the errors in our own? or just add additional logging. Thoughts?

Reply:
Since we dont have 'fail-fast' or 'optional' behavior, we dont need to swallow any server exceptions right now. Another reason we catch exception for list_configuration_settings and __resolve_keyvault_references would be during refresh operations. Since we dont have that either, we can leave those two for now.
I'd still expect to see our own error message if `json.loads` throws an error because we can add more specific details to the message to let users know exactly what failed.